App/Sagas/TextileSagas.js

```diff
@@ -239,25 +239,53 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
-    for (const photo of photos.reverse()) {
-      const multipartData = yield call(IPFS.addImageAtPath, photo.path, photo.thumbPath, 'default')
-      yield call(RNFS.unlink, photo.path)
-      yield call(RNFS.unlink, photo.thumbPath)
-      yield put(TextileActions.imageAdded('default', multipartData.boundary, multipartData.payloadPath))
-      yield put(IpfsNodeActions.getPhotoHashesRequest('default'))
-      yield call(
-        Upload.startUpload,
-        {
-          customUploadId: multipartData.boundary,
-          path: multipartData.payloadPath,
-          url: 'https://ipfs.textile.io/api/v0/add?wrap-with-directory=true',
-          method: 'POST',
-          type: 'multipart',
-          field: multipartData.boundary
+    const camera = yield select(TextileSelectors.camera)
+    let allPhotos = yield call(getAllPhotos)
+
+    // If camera.processed didn't exist, we'll add all but 1 photo to our
+    // ignore list and then set camera.processed through urisToIgnore
+    if (camera.processed === undefined) {
+      const ignoredPhotos = allPhotos.splice(1)
+      yield put(TextileActions.urisToIgnore(ignoredPhotos.map(photo => photo.uri)))
+    }
+
+    const processed = camera && camera.processed ? camera.processed : []
+    let allProcessed = processed.reduce((o, item, index) => ({...o, [item]: { index }}), {})
+    const photos = allPhotos.filter((photo) => {
+      if (allProcessed[photo.uri]) {
+        return false
+      }
+      return true
+    })
+
+    for (let photo of photos.reverse()) {
+      try { // single photo add and upload
+        photo = yield call(scalePhoto, photo)
+        const multipartData = yield call(IPFS.addImageAtPath, photo.path, photo.thumbPath, 'default')
+        yield put(TextileActions.imageAdded(photo.uri, 'default', multipartData.boundary, multipartData.payloadPath))
```

Comment:
You had originally asked that I join the photo.uri to be added to camera.processed in this single call @asutula, but that means it will be retried. sounds good but probably needs more thought when to retry. my gut says we actually want to add it to the processed in the event of errors, so in the finally, so that we don't retry it forever. if you think like this is better, all good, otherwise i can mod it quickly

Reply:
like this === your way, just add it when we do the imageAdded and retry on any errors

Reply:
Yea, so now if any error happens when processing the photo, we just 'forget' about the fact that we ever queried the photo and tried to process it. The next time `photosTask` runs, the failed photo will be fetched and processed again. Right?

I think that's fine for now. We can put more thought into a smarter system for retrying later on. It's not simple.

Reply:
yeah, that's right. cool